Formula/synergy-core.rb

```diff
@@ -0,0 +1,121 @@
+    depends_on "glib"
+    depends_on "libnotify"
+    depends_on "libxkbfile"
+  end
+
+  fails_with gcc: "5" do
+    cause "synergy-core requires C++17 support"
+  end
+
+  def install
+    # Use the standard brew installation path.
+    inreplace "CMakeLists.txt",
+              "set (SYNERGY_BUNDLE_DIR ${CMAKE_BINARY_DIR}/bundle)",
+              "set (SYNERGY_BUNDLE_DIR ${CMAKE_INSTALL_PREFIX}/bundle)"
+
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args,
+                    "-DBUILD_TESTS:BOOL=OFF", "-DCMAKE_INSTALL_DO_STRIP=1",
+                    "-DSYSTEM_PUGIXML:BOOL=ON"
+
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    if OS.mac?
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergy"  # main GUI program
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergys" # server
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergyc" # client
+    end
+  end
+
+  service do
+    run [opt_bin/"synergy"]
+    run_type :immediate
+  end
+
+  def caveats
+    # The official Synergy binaries built by Symless (for which it charges money)
+    # are signed by a trusted code-signing certificate. For that reason, the user
+    # only has to add Synergy.app to the 'Accessibility' permissions list once,
+    # and it will work forever, even after upgrading or overwriting the app (so
+    # long as the new versions are also signed by the same certificate).
+    #
+    # The binaries built by brew are not signed by a trusted certificate, so the
+    # experience may not be as smooth. The user may need to revoke all permissions
+    # for 'Accessibility' and re-grant them when upgrading synergy-core.
+    on_macos do
+      <<~EOS
+        On macOS, Synergy requires the 'Accessibility' permission.
+        You can grant this permission by navigating to:
+          System Preferences -> Security & Privacy -> Privacy -> Accessibility
+
+        If Synergy still doesn't work, try clearing the 'Accessibility' list:
+          sudo tccutil reset Accessibility
+        You can then grant the 'Accessibility' permission again.
+        You may need to clear this list each time you upgrade synergy-core.
+      EOS
+    end
+  end
+
+  test do
+    assert_equal shell_output("#{opt_bin}/synergys", 4),
+                 "synergys: no configuration available\n"
+    assert_equal shell_output("#{opt_bin}/synergyc", 3).split("\n")[0],
+                 "synergyc: a server address or name is required"
+    unless build.head?
+      version_string = Regexp.quote(version.to_s.split(".")[0..2].join("."))
+      assert_match(/synergys #{version_string}[\-.0-9a-z]*, protocol version/,
+                   shell_output("#{opt_bin}/synergys --version", 3).split("\n")[0])
+      assert_match(/synergyc #{version_string}[\-.0-9a-z]*, protocol version/,
+                   shell_output("#{opt_bin}/synergyc --version", 3).split("\n")[0])
+    end
```

Comment:
```suggestion
    return if head?

    version_string = Regexp.quote(version.to_s.split(".")[0..2].join("."))
    assert_match(/synergys #{version_string}[\-.0-9a-z]*, protocol version/,
                 shell_output("#{opt_bin}/synergys --version", 3).lines.first)
    assert_match(/synergyc #{version_string}[\-.0-9a-z]*, protocol version/,
                 shell_output("#{opt_bin}/synergyc --version", 3).lines.first)
```
Also, isn't
```ruby
version.to_s.split(".")[0..2].join(".")
```
just
```ruby
version.major_minor_patch
```
?